python/nvfuser/__init__.py

```diff
@@ -55,6 +56,63 @@ def disable_automatic_serialization():
     atexit.unregister(_C.serialize)
 
 
+# NOTE(crcrpar): The main motivation of this dataclass is to avoid unexpected negative values being supplied to indexing such as embedding.
+# See NVIDIA/Fuser#4529.
+class InputDescriptor:
+    """Dataclass to describe an input tensor."""
+
+    low: Number
+    high: Number
+    device: str
+    dtype: torch.dtype
+    size: tuple[int, ...]
+    strides: tuple[int, ...]
+    storage_offset: int
+    requires_grad: bool
+    is_contiguous: bool
+
+    def __init__(self, tensor: torch.Tensor) -> None:
+        if type(tensor) is not torch.Tensor:
+            msg = f"Repro script only supports {torch.Tensor} but {type(tensor)}"
+            raise RuntimeError(msg)
+        self.low, self.high = InputDescriptor._get_min_and_max(tensor)
+        self.dtype = tensor.dtype
+        self.device = f'"{str(tensor.device)}"'
+        self.size = tuple(tensor.size())
+        self.strides = tuple(tensor.stride())
+        self.storage_offset = tensor.storage_offset()
+        self.requires_grad = tensor.requires_grad
+        self.is_contiguous = tensor.is_contiguous()
+
+    @torch.inference_mode()
+    @staticmethod
+    def _get_min_and_max(tensor: torch.Tensor) -> tuple[Number | None, Number | None]:
+        if tensor.dtype is torch.bool:
+            return 0, 2
+
+        t = tensor
+        if t.dtype.is_floating_point and t.dtype.itemsize == 1:
+            t = t.float()
+
+        min_max = torch.aminmax(t)
+        return min_max[0].cpu().item(), min_max[1].cpu().item()
+
+    def make_repro_tensor(self) -> torch.Tensor:
```

Comment:
How is this used or tested?

Reply:
Could this function become `__call__`? Then, `InputDescriptor` becomes an `InputTensorFactory`.

Reply:
```python
factory = InputTensorFactory(tensor)
fake_input = factory()
run_something(fake_input)
```

Reply:
I like your proposal of making it `__call__` and the new name.
As for the use, yes, I think this can be YAGNI. I'll remove it for now. Once Thunder starts using this class, then there can be a need.